src/lib/components/FieldSelect.svelte

```diff
@@ -4,34 +4,55 @@
 	import type { LocalizedString } from 'typesafe-i18n';
 
 	import LL from '$i18n/i18n-svelte';
+	import Badge from '$lib/components/Badge.svelte';
 	import Button from '$lib/components/Button.svelte';
 	import Field from '$lib/components/Field.svelte';
 
 	export let name: string;
 	export let label: LocalizedString;
 	export let disabled: boolean | undefined = false;
-	export let options: Selected<string>[] = [];
+	export let options: OptionOrGroup[] = [];
 	export let value: string | undefined = undefined;
 	export let placeholder: string = '';
-	export let onChange: (value: Selected<string>) => void = () => {};
+	export let onChange: (value: Option) => void = () => {};
 
-	const noSelection = { value: '', label: '' };
+	type Option = Selected<string> & { extra?: string };
```

Comment:
Let's rename `extra` > `badge` so it's easier to _guess_ what component the value will be rendered in.